app/routes/__index/domains/index.tsx

```diff
@@ -1,22 +1,38 @@
 import { AddIcon } from '@chakra-ui/icons';
 import { Button, Container, Flex, Heading, Text } from '@chakra-ui/react';
+import type { LoaderArgs } from '@remix-run/node';
+import { Response } from '@remix-run/node';
 import { Link, useNavigate } from '@remix-run/react';
 import type { DomainsTableAction } from '~/components/domains-table';
 import DomainsTable from '~/components/domains-table';
-import DOMAINS_MOCK from '~/mocks/domains';
+import { getRecordsByUsername } from '~/models/record.server';
+import { getUser } from '~/session.server';
+import type { Record } from '@prisma/client';
+import { typedjson, useTypedLoaderData } from 'remix-typedjson';
+
+export const loader = async ({ request }: LoaderArgs) => {
+  const user = await getUser(request);
+  if (!user) {
+    throw new Response('No user', { status: 500 });
```

Comment:
@humphd not sure how I suppose to handle this case, just put it like that for now, I am curious what you think

Reply:
Don't use `getUser`, use https://github.com/DevelopingSpace/starchart/blob/main/app/session.server.ts#L48-L58 which will handle the redirect case if it's missing.